src/main/cc/wfa/virtual_people/common/field_filter/equal_filter.cc

```diff
@@ -0,0 +1,203 @@
+
+namespace wfa_virtual_people {
+
+// The implementation of EqualFilter based on the type of field represented by
+// @field_descriptors. The supported ValueTypes are
+//   int32_t
+//   int64_t
+//   uint32_t
+//   uint64_t
+//   bool
+//   const google::protobuf::EnumValueDescriptor*
+//   std::string
+template <typename ValueType>
+class EqualFilterImpl : public EqualFilter {
+ public:
+  explicit EqualFilterImpl(
+      std::vector<const google::protobuf::FieldDescriptor*>&& field_descriptors,
+      ValueType value):
+      field_descriptors_(std::move(field_descriptors)),
+      value_(value) {}
+
+  bool IsMatch(const google::protobuf::Message& message) const override;
+
+ private:
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors_;
+  ValueType value_;
+};
+
+template <>
+class EqualFilterImpl<std::string> : public EqualFilter {
+ public:
+  explicit EqualFilterImpl(
+      std::vector<const google::protobuf::FieldDescriptor*>&& field_descriptors,
+      const std::string& value):
+      field_descriptors_(std::move(field_descriptors)),
+      value_(value) {}
+
+  bool IsMatch(const google::protobuf::Message& message) const override;
+
+ private:
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors_;
+  std::string value_;
+};
+
+template <typename ValueType>
+bool EqualFilterImpl<ValueType>::IsMatch(
+    const google::protobuf::Message& message) const {
+  return value_ == GetValueFromProto<ValueType>(message, field_descriptors_);
+}
+
+template <>
+bool EqualFilterImpl<const google::protobuf::EnumValueDescriptor*>::IsMatch(
+    const google::protobuf::Message& message) const {
+  return (
+      value_->number() ==
+      GetValueFromProto<const google::protobuf::EnumValueDescriptor*>(
+          message, field_descriptors_)->number());
+}
+
+bool EqualFilterImpl<std::string>::IsMatch(
+    const google::protobuf::Message& message) const {
+  return (
+      value_ ==
+      GetValueFromProto<const std::string&>(message, field_descriptors_));
+}
+
+absl::StatusOr<std::unique_ptr<EqualFilter>> EqualFilter::New(
+    const google::protobuf::Descriptor* descriptor,
+    const FieldFilterProto& config) {
+  if (config.op() != FieldFilterProto::EQUAL) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Op must be EQUAL. Input FieldFilterProto: ", config.DebugString()));
+  }
+  if (!config.has_name()) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Name must be set. Input FieldFilterProto: ", config.DebugString()));
+  }
+  if (!config.has_value()) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Value must be set. Input FieldFilterProto: ", config.DebugString()));
+  }
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors;
+  absl::Status status =
+      GetFieldFromProto(descriptor, config.name(), &field_descriptors);
+  if (!status.ok()) {
+    return status;
```

Comment:
`RETURN_IF_ERROR(GetFieldFromProto(...))` could be simpler

Reply:
Done.

Reply:
i'm not seeing this change marked done. Am I missing something?

Reply:
Done. Sorry for missing this.